build.gradle

```diff
@@ -79,23 +79,23 @@ dependencies {
     compile 'postgresql:postgresql:9.1-901-1.jdbc4'
     compile 'com.github.insubstantial:substance:7.3'
     compile 'com.google.code.findbugs:jsr305:3.0.2'
-    compile 'com.google.guava:guava:19.0'
-    compile 'com.googlecode.soundlibs:jlayer:1.0.1-2'
-    compile 'com.sun.mail:mailapi:1.5.5'
-    compile 'com.sun.mail:smtp:1.5.5'
+    compile 'com.google.guava:guava:23.2-jre'
```

Comment:
There's several new Guava deprecation warnings:

```
src/main/java/games/strategy/engine/framework/systemcheck/LocalSystemChecker.java:40: warning: [deprecation] propagate(Throwable) in Throwables has been deprecated
        Throwables.propagate(e);
                  ^

src/main/java/games/strategy/engine/framework/systemcheck/LocalSystemChecker.java:50: warning: [deprecation] propagate(Throwable) in Throwables has been deprecated
        Throwables.propagate(e);
                  ^

src/main/java/games/strategy/util/UrlStreams.java:57: warning: [deprecation] propagate(Throwable) in Throwables has been deprecated
        throw Throwables.propagate(e);
                        ^

src/main/java/games/strategy/sound/ClipPlayer.java:313: warning: [deprecation] propagate(Throwable) in Throwables has been deprecated
      throw Throwables.propagate(e);
                      ^

src/test/java/games/strategy/engine/framework/systemcheck/SystemCheckTest.java:26: warning: [deprecation] propagate(Throwable) in Throwables has been deprecated
    final SystemCheck check = new SystemCheck("msg", () -> Throwables.propagate(testException));
                                                                     ^

src/test/java/games/strategy/engine/framework/systemcheck/SystemCheckTest.java:34: warning: [deprecation] propagate(Throwable) in Throwables has been deprecated
    final SystemCheck check = new SystemCheck("msg", () -> Throwables.propagate(testException));
                                                                     ^

src/test/java/games/strategy/engine/framework/systemcheck/LocalSystemCheckerTest.java:16: warning: [deprecation] propagate(Throwable) in Throwables has been deprecated
      new SystemCheck("throws exception", () -> Throwables.propagate(new Exception("test")));
                                                          ^

src/test/java/games/strategy/test/TestUtil.java:27: warning: [deprecation] write(CharSequence,File,Charset) in Files has been deprecated
      Files.write(contents, file, java.nio.charset.StandardCharsets.UTF_8);
           ^

src/test/java/games/strategy/test/TestUtil.java:30: warning: [deprecation] propagate(Throwable) in Throwables has been deprecated
      throw Throwables.propagate(e);
                      ^
```

Reply:
I actually saw them, but didn't realized they were coming from this PR ^^
Interesting article from the guava team: https://github.com/google/guava/wiki/Why-we-deprecated-Throwables.propagate